### DISCUSSION TITLE
Model without ancillary temperatures

### DISCUSSION THREAD

**Discussion by rohskopf** [wkirgsn/thermal-nn#6] [General]
Awesome project 👍

I am wondering if anyone has tried a TNN model without ancillary temperature inputs.

Consider a system where there are no ancillary temperatures that impact thermals - maybe just pure power sources and constant ambient temperature. Of course this is not the case in the given example, where coolant temperatures impact the system significantly.

Then the TNN would predict Delta T much more similarly to an RC network, but more flexible.

Fundamentally would this be a problem for TNN? Curious if anyone has tried.

**Comment by wkirgsn** [comment]
Hi, thank you!

We have experimented with less and more ancillary temperatures in the past.
This is easily done through the conductance matrix whose entries are the outputs of the gamma sub-NN.
From an implementation point of view, there are no problems at all.
However, modeling accuracy suffers, if there are significant heat sinks/sources not captured by a thermal node.

In case ambient components cannot be measured, you can also just add two constant heat sources -- a very cold and a very hot one.
Their temperature info is no use for the sub-NN inputs, however, there will be additional thermal conductances between target temperatures and those two dummy ancillary temperatures. These denote additional degrees of freedom during modeling and can improve accuracy quite significantly.

In the past, we also had added virtual temperatures, that are not measurable, but are modeled by the TNN anyway, such that it could impact the target temperatures in some way. That was not as constructive as having dummy ancillary temperatures though.
